common/src/test/java/com/microsoft/identity/common/AuthorityValidationMetadataCacheTest.java

```diff
@@ -0,0 +1,94 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common;
+
+import android.content.Context;
+import androidx.test.core.app.ApplicationProvider;
+import com.microsoft.identity.common.internal.cache.IAuthorityValidationMetadataCache;
+import com.microsoft.identity.common.internal.cache.SharedPreferencesAuthorityValidationMetadataCache;
+import com.microsoft.identity.common.java.exception.ClientException;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import static com.microsoft.identity.common.java.exception.ErrorStrings.ENVIRONMENT_CANNOT_BE_NULL_AS_A_AUTHORITY_VALIDATION_METADATA_KEY;
+import static com.microsoft.identity.common.java.exception.ErrorStrings.VALUE_CANNOT_BE_NULL_AS_A_AUTHORITY_VALIDATION_METADATA_VALUE;
+
+@RunWith(RobolectricTestRunner.class)
+public class AuthorityValidationMetadataCacheTest {
+
+    private IAuthorityValidationMetadataCache mAuthorityValidationMetadataCache;
+    private Context mContext;
+
+    @Before
+    public void setUp() throws Exception {
+        // Context and related init
+        mContext = ApplicationProvider.getApplicationContext();
+        mAuthorityValidationMetadataCache = new SharedPreferencesAuthorityValidationMetadataCache(mContext);
+    }
+
+    @After
+    public void tearDown() {
+        mAuthorityValidationMetadataCache.clearCache();
+    }
+
+    @Test
+    public void saveAuthorityValidationMetadataSuccess() throws ClientException {
+        String environment = "login.microsoftonline.com";
+        String metadata = "{metadata}";
+        String readMetadata = mAuthorityValidationMetadataCache.getAuthorityValidationMetadata(environment);
+        Assert.assertEquals(readMetadata, null);
+        mAuthorityValidationMetadataCache.saveAuthorityValidationMetadata(environment, metadata);
+        readMetadata = mAuthorityValidationMetadataCache.getAuthorityValidationMetadata(environment);
+        Assert.assertEquals(readMetadata, metadata);
+    }
+
+    @Test
+    public void readAuthorityValidationMetadataWithNullKey(){
+        try {
+            mAuthorityValidationMetadataCache.getAuthorityValidationMetadata(null);
+        }
+        catch (ClientException ex) {
+            Assert.assertEquals(ex.getErrorCode(), ENVIRONMENT_CANNOT_BE_NULL_AS_A_AUTHORITY_VALIDATION_METADATA_KEY);
+            return;
+        }
+        // Should not run to here.
+        Assert.assertTrue(false);
+    }
+
+    @Test

```

Comment:
```suggestion
    @Test(expected = IllegalArgumentException.class)

```

ProTip: there's a special syntax for testing exceptions are thown